File: src/contests/cg/puzzle/perl/the-descent.pl
# 🎮 CodinGame Puzzle - the-descent
# https://www.codingame.com/training/easy/the-descent

# Game loop: each turn, read 8 mountain heights and fire on the tallest one.
$| = 1;
while (1) {
    my $max_height = -1;
    my $max_index = 0;
    for my $i (0 .. 7) {
        my $h = <STDIN>;
        chomp $h;
        if ($h > $max_height) {
            $max_height = $h;
            $max_index = $i;
        }
    }
    print "$max_index\n";
}

File: src/contests/cg/puzzle/go/the-descent.go
// 🎮 CodinGame Puzzle - the-descent
// https://www.codingame.com/training/easy/the-descent

package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	// Game loop: each turn, read 8 mountain heights and fire on the tallest one.
	for {
		maxHeight, maxIndex := -1, 0
		for i := 0; i < 8; i++ {
			var h int
			fmt.Fscan(reader, &h)
			if h > maxHeight {
				maxHeight = h
				maxIndex = i
			}
		}
		fmt.Println(maxIndex)
	}
}
